src/chem/modis_aod2ioda.py

```diff
@@ -37,27 +37,38 @@
 
 AttrData = {
     'converter': os.path.basename(__file__),
+    'nvars': np.int32(len(obsvars)),
+}
+
+
+DimDict = {
+}
+
+VarDims = {
+    'aerosol_optical_depth': ['nlocs']
 }
 
 
 class AOD(object):
-    def __init__(self, filenames, obs_time, writer):
+    def __init__(self, filenames, obs_time):
         self.filenames = filenames
         self.obs_time = obs_time
-        self.writer = writer
         self.varDict = defaultdict(lambda: defaultdict(dict))
         self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-        self.loc_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-        self.var_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-        self.units = {}
+        self.varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
         self._read()
 
     def _read(self):
         # set up variable names for IODA
         for ncvar, iodavar in obsvars.items():
-            self.varDict[iodavar]['valKey'] = iodavar, self.writer.OvalName()
-            self.varDict[iodavar]['errKey'] = iodavar, self.writer.OerrName()
-            self.varDict[iodavar]['qcKey'] = iodavar, self.writer.OqcName()
+            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
```

Comment:
If the data provider has defined missing values for any of these variables, please specify those using the _FillValue variable attribute. Use "_FillValue" for the attribute name, and use the same data type as the variable for the attribute data type. Thanks!